### PR TITLE
Homework006

### PR DESCRIPTION
**Pull request by rodkevich**
If you wanna run task3 server - you should compile `homework/hw006/task03/server.go` to binary to make serving static files available.  
Running from IDE will lead to incorrect searching path
errChecks skipped for readability

### CODE REVIEW

File: homework/hw006/task02/internal/helpers/helpers.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+// MultiplyOrUpperDepOfType
+// 	Checks input if it’s an int, if true - returns input multiplied by 2
+// 	If it’s not an integer, return uppercase input string
+func MultiplyOrUpperDepOfType(items ...string) (rtn []string) {
+	for i := 0; i < len(items); i++ {
+		if _, errAtoi := strconv.Atoi(items[i]); errAtoi == nil {
+			fmt.Printf("%q was treated as a number\n", items[i])
+			var a int
+			_, errConv2Int := fmt.Sscanf(items[i], "%d", &a)
```
Comment (IgorZhadenovEpam) [inline homework/hw006/task02/internal/helpers/helpers.go:17]: Why do we do the int conversion twice?

File: homework/hw006/task02/internal/helpers/helpers.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package helpers
```
Comment (IgorZhadenovEpam) [inline homework/hw006/task02/internal/helpers/helpers.go:1]: I'm not sure if this simple function should be in a separate package

File: homework/hw006/task01/server.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+)
+
+type echoServer struct {
+	Address     string
+	somePrivate string
```
Comment (IgorZhadenovEpam) [inline homework/hw006/task01/server.go:12]: Для чего это?

Comment (rodkevich) [reply homework/hw006/task01/server.go:12]: для того чтобы научиться работать с приватными полями)

File: homework/hw006/task02/internal/helpers/helpers.go (hunk 1 above)
Comment (rodkevich) [reply homework/hw006/task02/internal/helpers/helpers.go:17]: Sure, it's not necessary. Just the way to remember both approaches of conversion

**Review by IgorZhadenovEpam — COMMENTED** [review]
Too much unnecessary code that is not relevant to the main task. We have said repeatedly that the final code should be clean, simple, readable, without any experimentation. As a result, we have a confusing implementation that should not be allowed in production. You can experiment in a separate branch.

File: homework/hw006/task02/client.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+// makeCallToServer function connects and requests remote
+func (c clientCLI) makeCallToServer(lines []byte) {
+
+	url := "http://" + c.Address
+	body := bytes.NewBuffer(lines)
+	req, _ := http.NewRequest(http.MethodPost, url, body)
+	res, _ := c.client.Do(req)
+	rbd, _ := ioutil.ReadAll(res.Body)
```
[the remark this answers is not in the document]
Comment (IgorZhadenovEpam) [reply homework/hw006/task02/client.go]: oops sorry i missed

**Comment by rodkevich** [comment]
unnecessary fields and type conversion where removed

File: homework/hw006/task01/server.go (hunk 3 above)
Comment (rodkevich) [reply homework/hw006/task01/server.go]: removed

**Review by IgorZhadenovEpam — APPROVED** [review]

**Review by DzmitryYafremenka — APPROVED** [review]

**Pull request merged**